src/tools/perf/lib/Makefile.am

```diff
@@ -18,6 +18,24 @@ libucxperf_la_LIBADD   = \
 	$(abs_top_builddir)/src/ucp/libucp.la \
 	$(abs_top_builddir)/src/ucs/libucs.la
 
+# Add CUDA dependencies
+if HAVE_CUDA
+libucxperf_la_CPPFLAGS += $(CUDA_CPPFLAGS)
+libucxperf_la_CFLAGS   += $(CUDA_CFLAGS)
+libucxperf_la_LDFLAGS  += $(CUDA_LDFLAGS) -version-info $(SOVERSION)
+libucxperf_la_LIBADD   += $(CUDART_LIBS)
+endif
+
+# Add ROCM dependencies
+if HAVE_HIP
+libucxperf_la_CPPFLAGS += $(HIP_CPPFLAGS)
+libucxperf_la_CFLAGS   += $(HIP_CFLAGS)
+libucxperf_la_LDFLAGS  += $(HIP_LDFLAGS) $(HIP_LIBS) -version-info $(SOVERSION) \
+                          $(patsubst %, -Xlinker %, -L$(ROCM_ROOT)/lib -rpath $(ROCM_ROOT)/hip/lib -rpath $(ROCM_ROOT)/lib) \
+                          $(patsubst %, -Xlinker %, --enable-new-dtags) \
+                          $(patsubst %, -Xlinker %, -rpath $(ROCM_ROOT)/lib64)
+endif
+
```

Comment:
we cannot add direct dependency on cuda/rocm because then if UCX is built with cuda support but used on a system without cuda libs, it will not work

Reply:
So `libperf_memcopy.c` should be a module?